library/core/src/hint.rs

```diff
@@ -219,6 +219,75 @@ pub fn spin_loop() {
 /// backend used. Programs cannot rely on `black_box` for *correctness* in any way.
 ///
 /// [`std::convert::identity`]: crate::convert::identity
+///
+/// # When is this useful?
+///
+/// First and foremost: `black_box` does _not_ guarantee any exact behavior and, in some cases, may
+/// do nothing at all. As such, it **must not be relied upon to control critical program behavior.**
+/// This _immediately_ precludes any direct use of this function for cryptographic or security
+/// purposes.
+///
+/// While not suitable in those mission-critical cases, `back_box`'s functionality can generally be
+/// relied upon for benchmarking, and should be used here. Its purpose is to help ensure that the
```

Comment:
```suggestion
/// relied upon for benchmarking, and should be used there. It will try to ensure that the
```
Stuff like this always reminds me of how crude my knowledge of English is even as a native speaker. I don't know if "here" or "there" is more correct, but "there" sounds better, because this comes directly after a general mention of benchmarking not a specific call to `hint::black_box`.

And "help ensure" just puts together an attempt and a guarantee so it reads awkward to me. (I don't know if I originally suggested this particular phrasing that I am now criticizing. I might have :joy: )